pkg/importer/http-datasource.go

```diff
@@ -133,20 +133,20 @@ func (hs *HTTPDataSource) Info() (ProcessingPhase, error) {
 	if hs.contentType == cdiv1.DataVolumeArchive {
 		return ProcessingPhaseTransferDataDir, nil
 	}
-	if !hs.readers.Convert {
-		return ProcessingPhaseTransferDataFile, nil
-	}
-	if pullMethod, _ := util.ParseEnvVar(common.ImporterPullMethod, false); pullMethod == string(cdiv1.RegistryPullNode) {
-		hs.url, _ = url.Parse(fmt.Sprintf("nbd+unix:///?socket=%s", nbdkitSocket))
-		if err = hs.n.StartNbdkit(hs.endpoint.String()); err != nil {
-			return ProcessingPhaseError, err
+	if hs.readers.Convert {
+		if hs.brokenForQemuImg || hs.readers.Archived || hs.customCA != "" {
+			return ProcessingPhaseTransferScratch, nil
 		}
-		return ProcessingPhaseConvert, nil
+	} else {
+		if hs.readers.Archived || hs.customCA != "" {
+			return ProcessingPhaseTransferDataFile, nil
```

Comment:
Should return ProcessingPhaseTransferScratch here if we want to have proper preallocation

Reply:
I saw this comment earlier and understood the opposite https://github.com/kubevirt/containerized-data-importer/pull/2832#discussion_r1283698347, should we return transferScratch when preallocation is false or true? Anyway, I'm thinking of simplifying this PR to just address this bug so we can backport it safely and reconsider the usage of nbdkit again in a follow-up.